### PR TITLE
feat(frontend): Conditionally enable to override the existing pipeline name with the name of uploaded file

### PR DESCRIPTION
**Pull request by jlyaoyuli**
The main purpose of this PR is adding a checkbox to determine if the existing pipeline name is replaced with the name of uploaded file.

Before:

https://user-images.githubusercontent.com/56132941/216728770-c9b185a7-fd81-44fa-8a45-1b14ba2ea080.mov

After:

https://user-images.githubusercontent.com/56132941/217395164-050bd86c-fa0f-40ff-ade5-9ff509ba2f12.mov

### CODE REVIEW

**Comment by chensun** [comment]
@jlyaoyuli Actually, I wonder if the existing behavior--fill in pipeline name based on file name only when it's empty--is okay and even safer to the user. Consider a user follow the order of inputs, and entered the pipeline name by themself, then later they choose a file to upload, and that file name overrides the previously user provided pipeline name. Which doesn't seems like a common practice to me. WDYT?

/cc @zijianjoy for a second opinion as well.

**Comment by zijianjoy** [comment]
Thank you Joe for the PR! I understand the desire to keep the pipeline name consistent with file name in this PR.

I tend to agree with Chen that we want to avoid accidentally overriding user input when they switch to a different file, especially when the `Pipeline Name` input box appears earlier than file selector in the UI.

There are two options I would like to propose, I would like to hear your opinion:

1. We can introduce a checkbox to `override pipeline name`: if this box is checked, we always override `Pipeline Name` with file name. Otherwise, we fill in `Pipeline Name` only when this is empty.
2. Open a dialog to ask users whether they want to override `Pipeline Name` with file name, after user chooses a file to upload.

**Comment by jlyaoyuli** [comment]
> 

Thank you @chensun and @zijianjoy for your suggestions!
Yes, I agree that The pipeline name is a field that users might defined before uploading the file. Directly override the name possibly bring out some inconveniences. Preserving the pipeline name makes sense to me and I tend to choose option 1 to implement!

File: frontend/src/pages/NewPipelineVersion.tsx [hunk 1]
```diff
@@ -437,7 +451,7 @@ class NewPipelineVersion extends Page<{}, NewPipelineVersionState> {
               // Find a better to align this input box with others
               style={{
                 maxWidth: 2000,
-                width: 465,
```
[the remark this answers is not in the document]
Comment (jlyaoyuli) [reply frontend/src/pages/NewPipelineVersion.tsx:440]: Screenshot 2023-02-08 at 11 09 34 AM

I think the original one is a little shorter than others.

[the remark this answers is not in the document]
Comment (jlyaoyuli) [reply frontend/src/pages/NewPipelineVersion.tsx:440]: After changing the width:

Screenshot 2023-02-08 at 11 27 34 AM

**Comment by jlyaoyuli** [comment]
Close first and wait for more discussion

**Pull request closed without merging**